tests/plan/selectors_test.py

```
@@ -0,0 +1,393 @@
+    # NOTE: Runtime isn't as strict as the annotation which requires instantiated `DType`
+    return nw.Schema(mapping)  # type: ignore[arg-type]
+
+
+@pytest.fixture(scope="module")
+def schema_nested_1() -> nw.Schema:
+    return _schema(
+        {
+            "a": nw.Int32(),
+            "b": nw.List(nw.Int32()),
+            "c": nw.List(nw.UInt32),
+            "d": nw.Array(nw.Int32, 3),
+            "e": nw.List(nw.String),
+            "f": nw.Struct({"x": nw.Int32}),
+        }
+    )
+
+
+@pytest.fixture(scope="module")
+def schema_nested_2() -> nw.Schema:
+    return _schema(
+        {
+            "a": nw.Int32(),
+            "b": nw.Array(nw.Int32, 4),
+            "c": nw.Array(nw.UInt32, 4),
+            "d": nw.Array(nw.Int32, 3),
+            "e": nw.List(nw.Int32),
+            "f": nw.Array(nw.String, 4),
+            "g": nw.Struct({"x": nw.Int32}),
+        }
+    )
+
+
+@pytest.fixture(scope="module")
+def schema_non_nested() -> nw.Schema:
+    return _schema(
+        {
+            "abc": nw.UInt16(),
+            "bbb": nw.UInt32(),
+            "cde": nw.Float64(),
+            "def": nw.Float32(),
+            "eee": nw.Boolean(),
+            "fgg": nw.Boolean(),
+            "ghi": nw.Time(),
+            "JJK": nw.Date(),
+            "Lmn": nw.Duration(),
+            "opp": nw.Datetime("ms"),
+            "qqR": nw.String(),
+        }
+    )
+
+
+class Frame:
+    def __init__(self, schema: nw.Schema) -> None:
+        self.schema = schema
+        self.columns = tuple(schema.names())
+
+    @staticmethod
+    def from_mapping(mapping: IntoSchema) -> Frame:
+        return Frame(nw.Schema(mapping))
+
+    @staticmethod
+    def from_names(*column_names: str) -> Frame:
+        """Construct with all `nw.Int64()`."""
+        return Frame(nw.Schema((name, nw.Int64()) for name in column_names))
+
+    @property
+    def width(self) -> int:
+        return len(self.columns)
```

Needs coverage, the test that calls it is failing early and doesn't reach it